packages/theme-patternfly-org/scripts/md/mdx-hast-to-jsx.js

```diff
@@ -0,0 +1,185 @@
+  }
+}
+
+function serializeRoot(node, options) {
+  const { getRelPath, getPageData } = options;
+  const pageData = {...getPageData()};
+  const exportName = pageData.slug.replace(/[\/-](.)?/g, (_, match) => capitalize(match)) + 'Docs';
+  options.exportName = exportName;
+  const groups = {
+    import: [],
+    rest: []
+  };
+
+  for (const child of node.children) {
+    let kind = 'rest'
+
+    if (child.type === 'import') {
+      kind = child.type
+    }
+
+    groups[kind].push(child)
+  }
+
+  const importStatements = groups.import
+    .map(childNode => toJSX(childNode, node))
+    .map(imp => imp.replace(/(['"])\./g, (_, match) => `${match}${getRelPath()}`))
+    .join('\n')
+
+  // https://astexplorer.net/#/gist/9c531dd372dfc57e194c13c2889d31c3/03f2d6e889db1a733c6a079554e8af7784863739
+  const importSpecifiers = jsxParser.parse(importStatements, { sourceType: 'module' }).body
+    .map(node => node.specifiers)
+    .flat(1)
+    .map(spec => spec.local ? spec.local.name : null)
+    .filter(localName => !/srcImport\d+/.test(localName)) // Images in MD like [!img](./src)
+    .filter(Boolean)
+    .join(',\n  ');
+
+  const childNodes = groups.rest
+    .map(childNode => toJSX(childNode, node, options))
+    .join('');
+
+  return `import React from 'react';
+import { AutoLinkHeader, Example, Link as PatternflyThemeLink } from 'theme-patternfly-org/components';
+${importStatements}
+
+export const ${exportName} = ${JSON.stringify(pageData, null, 2)};
+${exportName}.liveContext = {
+  ${importSpecifiers}
+};
+${exportName}.Component = () => (
+  <React.Fragment>${childNodes.replace(/\n\s*\n/g, '\n')}
+  </React.Fragment>
+);
+`;
+}
+
+function serializeElement(node, options) {
+  const { indent, exportName } = options;
+  const {type, props} = toH(
+    fakeReactCreateElement,
+    Object.assign({}, node, {children: []}),
+    {prefix: false}
+  );
+  const content = serializeChildren(node, {...options});
+
+  delete props.key;
+
+  const srcImport = node.properties.src;
+  if (type === 'img') {
+    delete props.src;
+  }
+  const spread = Object.keys(props).length === 0
+    ? null
+    : JSON.stringify(props);
+
+  const indentText = '  '.repeat(indent);
+
+  return `
+${indentText}<${type}${
+  type === 'Example' ? ` {...${exportName}}` : ''}${
+  type === 'img' && srcImport ? ` src={${srcImport}}` : ''}${
+  spread ? ` {...${spread}}` : ''}>
+${indentText}  ${content}
+${indentText}</${type}>`;
+}
+
+function serializeComponent(node, options) {
+  let content = serializeChildren(node, options);
+  const tags = serializeTags(
+    Object.assign({}, node, {children: content ? ['!'] : []})
+  );
+
+  if (node.type === 'mdxBlockElement' && content) {
+    content = '\n' + content + '\n';
+  }
+
+  const indentText = '  '.repeat(options.indent);
+  return `${indentText}${tags.open}${content}${tags.close || ''}`;
+}
+
+function serializeText(node, options, parentNode) {
+  const preserveNewlines = options.preserveNewlines;
```

Comment:
nit: probably unnecessary to save to a variable since it's only used once

Reply:
This code was copied. Will remove for brevity's sake.